apps/teams/tests/test_backends.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+
+from apps.experiments.models import Experiment
+from apps.teams.backends import TeamBackend
+from apps.teams.utils import current_team
+from apps.utils.factories.team import MembershipFactory
+from apps.utils.factories.user import GroupFactory
+
+
+@pytest.fixture()
+def experiment_content_type():
+    return ContentType.objects.get_for_model(Experiment)
+
+
+@pytest.fixture()
+def experiment_permissions(experiment_content_type):
+    return {perm.codename: perm for perm in Permission.objects.filter(content_type=experiment_content_type)}
+
+
+@pytest.fixture()
+def group1(experiment_permissions):
+    group = GroupFactory()
+    group.permissions.add(experiment_permissions["add_experiment"], experiment_permissions["change_experiment"])
+    return group
+
+
+@pytest.fixture()
+def group2(experiment_permissions):
+    group = GroupFactory()
+    group.permissions.add(experiment_permissions["view_experiment"], experiment_permissions["delete_experiment"])
+    return group
+
+
+@pytest.fixture(autouse=True)
+def use_team_backend(settings):
+    settings.AUTHENTICATION_BACKENDS = ["apps.teams.backends.TeamBackend"]
+
+
+@pytest.mark.django_db()
+def test_team_backend_no_current_team(group1, group2):
```

Comment:
The usge of the context variable made this test a bit unintuitive. My first thought was that the user should have had permissions, since
1. the user did have team membership, and
2. the membership did belong to groups with permissions.

Keeping in mind that the context variable was not set, this test makes sense. Might be good to add a comment about that

Reply:
362252fa0543eb9177242fcf450882f6f475c2f2